paimon-core/src/main/java/org/apache/paimon/table/system/ManifestsTable.java

```diff
@@ -198,6 +199,7 @@ private static List<ManifestFileMeta> allManifests(FileStoreTable dataTable) {
         CoreOptions coreOptions = CoreOptions.fromMap(dataTable.options());
         SnapshotManager snapshotManager = dataTable.snapshotManager();
         Long snapshotId = coreOptions.scanSnapshotId();
+        String tagName = coreOptions.scanTagName();
```

Comment:
Here add a check, if both snapshotId and tagName are not null, then there is ambiguity here.

Reply:
tagName had checked follow line：if (!StringUtils.isEmpty(tagName) && dataTable.tagManager().tagExists(tagName))